packages/beacon-node/src/chain/validation/dataColumnSidecar.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH,
+  KZG_COMMITMENTS_SUBTREE_INDEX,
+  DATA_COLUMN_SIDECAR_SUBNET_COUNT,
+  NUMBER_OF_COLUMNS,
+} from "@lodestar/params";
+import {ssz, deneb, peerdas, Slot, Root} from "@lodestar/types";
+import {verifyMerkleBranch} from "@lodestar/utils";
+
+import {DataColumnSidecarGossipError, DataColumnSidecarErrorCode} from "../errors/dataColumnSidecarError.js";
+import {GossipAction} from "../errors/gossipValidation.js";
+import {IBeaconChain} from "../interface.js";
+
+export async function validateGossipDataColumnSidecar(
+  chain: IBeaconChain,
+  dataColumnSideCar: peerdas.DataColumnSidecar,
+  gossipIndex: number
+): Promise<void> {
+  const dataColumnSlot = dataColumnSideCar.signedBlockHeader.message.slot;
+
+  if (
+    dataColumnSideCar.index > NUMBER_OF_COLUMNS ||
+    dataColumnSideCar.index % DATA_COLUMN_SIDECAR_SUBNET_COUNT !== gossipIndex
+  ) {
+    throw new DataColumnSidecarGossipError(GossipAction.REJECT, {
+      code: DataColumnSidecarErrorCode.INVALID_INDEX,
+      columnIndex: dataColumnSideCar.index,
+      gossipIndex,
+    });
+  }
+
+  // [IGNORE] The sidecar is not from a future slot (with a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) --
+  // i.e. validate that sidecar.slot <= current_slot (a client MAY queue future blocks for processing at
+  // the appropriate slot).
+  const currentSlotWithGossipDisparity = chain.clock.currentSlotWithGossipDisparity;
+  if (currentSlotWithGossipDisparity < dataColumnSlot) {
+    throw new DataColumnSidecarGossipError(GossipAction.IGNORE, {
+      code: DataColumnSidecarErrorCode.FUTURE_SLOT,
+      currentSlot: currentSlotWithGossipDisparity,
+      blockSlot: dataColumnSlot,
+    });
+  }
+
+  validateInclusionProof(dataColumnSideCar);
```

Comment:
Just noticed that this returns a bool, not throws for invalid proof.  Should there be a check and error thrown here?

Reply:
yes please

Reply:
i have added this, can you add more gossip validations and errors on the lines of blobsidecar (check blobsider car gossip validation file)